src/Microsoft.ML.Data/Transforms/KeyToVectorTransform.cs

```diff
@@ -69,530 +75,776 @@ public bool TryUnparse(StringBuilder sb)
-            Host.Assert(Infos[iinfo].TypeSrc.IsVector);
-            Host.Assert(Infos[iinfo].TypeSrc.ItemType.IsKey);
-            Host.Assert(!_bag[iinfo]);
-
-            var info = Infos[iinfo];
-            int size = info.TypeSrc.ItemType.KeyCount;
-            Host.Assert(size > 0);
-
-            int cv = info.TypeSrc.VectorSize;
-            Host.Assert(cv >= 0);
-            Host.Assert(_types[iinfo].VectorSize == size * cv);
-
-            var getSrc = RowCursorUtils.GetVecGetterAs<uint>(NumberType.U4, input, info.Source);
-            var src = default(VBuffer<uint>);
-            return
-                (ref VBuffer<Float> dst) =>
-                {
-                    getSrc(ref src);
-                    int lenSrc = src.Length;
-                    Host.Check(lenSrc == cv || cv == 0);
-
-                    // Since we generate values in order, no need for a builder.
-                    var valuesDst = dst.Values;
-                    var indicesDst = dst.Indices;
-
-                    int lenDst = checked(size * lenSrc);
-                    int cntSrc = src.Count;
-                    if (Utils.Size(valuesDst) < cntSrc)
-                        valuesDst = new Float[cntSrc];
-                    if (Utils.Size(indicesDst) < cntSrc)
-                        indicesDst = new int[cntSrc];
-
-                    var values = src.Values;
-                    int count = 0;
-                    if (src.IsDense)
-                    {
-                        Host.Assert(lenSrc == cntSrc);
-                        for (int slot = 0; slot < cntSrc; slot++)
-                        {
-                            Host.Assert(count < cntSrc);
-                            uint key = values[slot] - 1;
-                            if (key >= (uint)size)
-                                continue;
-                            valuesDst[count] = 1;
-                            indicesDst[count++] = slot * size + (int)key;
-                        }
-                    }
-                    else
-                    {
-                        var indices = src.Indices;
-                        for (int islot = 0; islot < cntSrc; islot++)
-                        {
-                            Host.Assert(count < cntSrc);
-                            uint key = values[islot] - 1;
-                            if (key >= (uint)size)
-                                continue;
-                            valuesDst[count] = 1;
-                            indicesDst[count++] = indices[islot] * size + (int)key;
-                        }
-                    }
-                    dst = new VBuffer<Float>(lenDst, count, valuesDst, indicesDst);
-                };
+            Contracts.CheckValue(input, nameof(input));
+            return new OutVectorColumn<TKey, TValue>(input, new Config(bag));
         }
+
+        /// <summary>
+        /// Takes a column of key type of known cardinality and produces an indicator vector of floats.
```

Comment:
>Takes a column of key type of known cardinality and produces an indicator vector of floats. [](start = 12, length = 91)

This description is insufficient. Do you feel like you could write better documentation, more in line with the other Pigsty extension methods? It is important for our public API that we get that right.

A sufficient description would describe what is happening to the data. Also crucial is a discussion on what happens in bagged vs. not.

Reply:
Is this better?

---
In reply to: [216801051](https://github.com/dotnet/machinelearning/pull/858#discussion_r216801051) [](ancestors = 216801051)